### PR TITLE
Group notifications and add avatars

### PR DESCRIPTION
**Pull request by Macludde**
Add functionality (from previous version of site) to group together similar notifications for the same resource, i.e. likes on the same article, pings etc.

Also shows avatars in notification window

### CODE REVIEW

File: src/lib/server/shop/types.ts [hunk 1]
```diff
@@ -22,4 +22,5 @@ export const dbIdentification = ({
   memberId ? { memberId } : { externalCustomerCode: externalCode };
 
 export const GRACE_PERIOD_WINDOW = 60 * 1000; // 1 minute
-export const TIME_TO_BUY = 5 * 60 * 1000; // 5 minutes
+// export const TIME_TO_BUY = 5 * 60 * 1000; // 5 minutes
+export const TIME_TO_BUY = 10 * 1000; // 5 minutes
```
Comment (danieladugyan) [inline src/lib/server/shop/types.ts:26]: Remove one of them

File: src/lib/utils/notifications/group.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        member: Member;
+        mandate:
+          | (Mandate & {
+              position: Position;
+            })
+          | null;
+        customAuthor: CustomAuthor | null;
+      })
+    | null;
+};
```
Comment (danieladugyan) [inline src/lib/utils/notifications/group.ts:28]: If we can use [this pattern instead](https://www.prisma.io/docs/orm/prisma-client/type-safety/operating-against-partial-structures-of-model-types#solution-1), I'd prefer that.

File: src/routes/(app)/api/notifications/my/+server.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { groupNotifications } from "$lib/utils/notifications/group";
+import type { PrismaClient } from "@prisma/client";
+import type { AuthUser } from "@zenstackhq/runtime";
+import type { RequestHandler } from "./$types";
+
+/**
+ * Get's all notifications for a user
```
Comment (danieladugyan) [inline src/routes/(app)/api/notifications/my/+server.ts:7]:
```suggestion
 * Gets all notifications for a user
```

**Review by danieladugyan — APPROVED** [review]
Other than the small comments, LGTM

File: src/lib/server/shop/types.ts (hunk 1 above)
Comment (Macludde) [reply src/lib/server/shop/types.ts:26]: No, I like redundancy

File: src/lib/utils/notifications/group.ts (hunk 2 above)
Comment (Macludde) [reply src/lib/utils/notifications/group.ts:28]: Aww nice, was looking for something like that!

File: src/routes/(app)/api/notifications/my/+server.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  });
+};
+
+export const getMyGroupedNotifications = async (
+  user: AuthUser,
+  prisma: PrismaClient,
+) => {
+  const myNotifications = await getMyNotifications(user, prisma);
+  return groupNotifications(myNotifications);
+};
```
Comment (danieladugyan) [inline src/routes/(app)/api/notifications/my/+server.ts:42]: Unfortunately, you are not allowed to export anything other than the SvelteKit "magic" functions from a +server.ts file (ile GET). pls fix :smile: @Macludde

Comment (Macludde) [reply src/routes/(app)/api/notifications/my/+server.ts:42]: Oh oki

Comment (Macludde) [reply src/routes/(app)/api/notifications/my/+server.ts:42]: Fixed in #409

**Pull request merged**